protocol/communities/manager.go

```diff
@@ -1007,19 +1041,26 @@ func (m *Manager) ImportCommunity(key *ecdsa.PrivateKey) (*Community, error) {
 		}
 
 		config := Config{
-			ID:                   &key.PublicKey,
-			PrivateKey:           key,
+			ID:                   &communityKey.PublicKey,
+			PrivateKey:           communityKey,
 			Logger:               m.logger,
 			Joined:               true,
 			MemberIdentity:       &m.identity.PublicKey,
 			CommunityDescription: description,
+			Shard:                importData.Shard,
 		}
 		community, err = New(config)
 		if err != nil {
 			return nil, err
 		}
 	} else {
-		community.config.PrivateKey = key
+		community.config.PrivateKey = communityKey
+		community.config.Shard = importData.Shard
+	}
+
+	err = m.transport.StorePubsubTopicKey(community.PubsubTopic(), protectedTopicKey)
```

Comment:
Would it make sense here to check if `protectedTopicKey` is not nil first, before calling `StorePubsubTopicKey`? The reason is that `protectedTopicKey` is only set if `importData.TopicKey` is present, hence if `TopicKey` is nil we will call `StorePubsubTopicKey` with a nil private key.